src/coreclr/jit/emitarm64.cpp

```diff
@@ -13597,6 +13597,33 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
         // src2 can only be a reg
         assert(!src2->isContained());
     }
+    else if ((src1->OperIs(GT_MUL) && src1->isContained()) || (src2->OperIs(GT_MUL) && src2->isContained()))
```

Comment:
Isn't the `src1->gtGetOp1()` or `src1->gtGetOp2()` would be the one that is marked contained in lower and that should be checked here?

Reply:
`madd` fot `GT_ADD` can only be emitted when GT_MUL is contained. So "isContaind" here basically means lower approved this tree to be optimized into madd. GT_MUL can be not contained e.g. if it has gtOverflow set